### PR TITLE
[release/v2.22] Bump machine-controller to v1.56.3 and KubeOne to 1.6.2

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:

This bumps machine-controller to v1.56.3. Its changelog:

> ### Bugfixes
> 
> - Allow external CCMs to handle failing node objects before MC ([#1652](https://github.com/kubermatic/machine-controller/pull/1652))
> - Anexia Provider: Fix nil pointer exception when providing no network configuration ([#1650](https://github.com/kubermatic/machine-controller/pull/1650))
> 
> ### New Feature
> 
> - Anexia Provider: Enable up to 4 custom dns entries ([#1640](https://github.com/kubermatic/machine-controller/pull/1640))
> 
> ### Misc
> 
> - Machine-controller is now built with Go 1.19.9 ([#1634](https://github.com/kubermatic/machine-controller/pull/1634))

KubeOne changelog at https://github.com/kubermatic/kubeone/releases/tag/v1.6.2.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
- Update machine-controller to [v1.56.3](https://github.com/kubermatic/machine-controller/releases/tag/v1.56.3)
- Update KubeOne to [v1.6.2](https://github.com/kubermatic/kubeone/releases/tag/v1.6.2)
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: go.mod
Comment (xmudrii) [inline go.mod:1]: Should we also bind KubeOne to the stable version here?

Comment (embik) [reply go.mod:1]: Done, good catch!

**Review by xmudrii — APPROVED** [review]

**Pull request merged**